### PR TITLE
[DA-2856] Genomic Scheduling API

### PR DESCRIPTION
**Pull request by dtharpeuno**
## Resolves *[ticket DA-2856]*
https://precisionmedicineinitiative.atlassian.net/browse/DA-2856

## Description of changes/additions
Adding genomic scheduling API for partner usage for sending notifications when appointments with genomic counselors have been scheduled

## Tests
- [x] unit tests

### CODE REVIEW

File: rdr_service/dao/genomics_dao.py [hunk 1]
```diff
@@ -2213,6 +2213,106 @@ def get_report_state_query_config(self):
...
+                and_(
+                    note_alias.appointment_id == GenomicAppointmentEvent.appointment_id,
+                    note_alias.event_type.like('%note_available')
+                )
+            ).filter(
+                and_(
+                    GenomicAppointmentEvent.appointment_id == max_appointment_id_subquery.c.max_appointment_id,
+                    GenomicAppointmentEvent.event_authored_time ==
+                    max_event_authored_time_subquery.c.max_event_authored_time
+                )
```
Comment (j-kanuch) [inline rdr_service/dao/genomics_dao.py:2296]: Do these conditions work if more than one row is returned in the sub-query?  I'm more familiar with using raw SQL and doing an inner join to the sub-query, but it may be handled differently via the ORM.

Comment (dtharpeuno) [reply rdr_service/dao/genomics_dao.py:2296]: Good question, yes they do I added unit tests to encompass all the usual edge cases to test the queries correctly. Also for each query change, I extracted and ran the corresponding RAW SQL output to make sure the query was being translated correctly from the ORM.

Comment (j-kanuch) [reply rdr_service/dao/genomics_dao.py:2296]: I see you're testing multiple participants in `test_pass_start_date_params`.

**Review by j-kanuch — APPROVED** [review]

**Review by kenny-skaggs — APPROVED** [review]
